supervision/annotators/core.py

```diff
@@ -85,6 +85,118 @@ def annotate(self, scene: np.ndarray, detections: Detections) -> np.ndarray:
         return scene
 
 
+class HeatmapAnnotator:
+    """
+    A class for drawing heatmap on an image using provided detections.
+    Heat is accumulated over time. Drawn as a semi-transparent overlay as blured circles.
+    """
+
+    def __init__(
+        self,
+        position: Optional[Position] = Position.BOTTOM_CENTER,
+        opacity: float = 0.2,
+        radius: int = 40,
+        kernel_size: Optional[int] = 25,
+    ):
+        """
+        Args:
+            position (Optional[Position]): The position of the heatmap.
+                Defaults to `BOTTOM_CENTER`.
+            opacity (float): Opacity of the overlay mask. Must be between `0` and `1`.
+            radius (int): Radius of the heat circle.
+            kernel_size (Optional[int]): Kernel size for blurring the heatmap.
+        """
+        self.position = position
+        self.opacity = opacity
+        self.radius = radius
+        self.kernel_size = kernel_size
+        self.heatmask = None
+
+    def get_xy(
```

Comment:
Hi @kapter 👋🏻! I think we already have that logic implemented: https://github.com/roboflow/supervision/blob/9ded70c8daae214e53cb7b33559c50817af14e68/supervision/detection/core.py#L603

Reply:
It's my mistake, I saw same logic in LabelAnnotator and repeat it but now I understood why.
I can create new commit tonight.